cmd/activator/main.go

```diff
@@ -199,12 +206,52 @@ func main() {
 		return revisionInformer.Lister().Revisions(revID.Namespace).Get(revID.Name)
 	}
 
+	activationResultGetter := func(revID activator.RevisionID) *activator.ActivationResult {
+		rev, err := revisionGetter(revID)
+		if err != nil {
+			return getActivationResultFromError(err)
+		}
+		serviceName, configurationName := getLabels(rev)
+
+		name := revisionresourcenames.K8sService(rev)
+		svc, err := serviceInformer.Lister().Services(revID.Namespace).Get(name)
+		if err != nil {
+			return getActivationResultFromError(err)
+		}
+
+		fqdn := fmt.Sprintf("%s.%s.svc.%s", name, rev.Namespace, utils.GetClusterDomainName())
+
+		// Search for the correct port in all the service ports.
+		port := int32(-1)
+		for _, p := range svc.Spec.Ports {
+			if p.Name == revisionresources.ServicePortName(rev) {
+				port = p.Port
+				break
+			}
+		}
+		if port == -1 {
+			return &activator.ActivationResult{
+				Status: http.StatusInternalServerError,
+				Error:  errors.New("revision needs external HTTP port"),
+			}
+		}
+
+		endpoint := activator.Endpoint{fqdn, port}
+
+		return &activator.ActivationResult{
+			Endpoint:          endpoint,
+			ServiceName:       serviceName,
+			ConfigurationName: configurationName,
```

Comment:
What's this used for?

Reply:
It's used in stats reporting in the Handler.